src/SourceNodes/hlsnode.ts

```diff
@@ -0,0 +1,159 @@
+        // when calling _load and improve performance
+        const maxBufferLength =
+            duration === undefined
+                ? DEFAULT_MAX_BUFFER_LENGTH
+                : duration < DEFAULT_MAX_BUFFER_LENGTH
+                    ? duration
+                    : DEFAULT_MAX_BUFFER_LENGTH;
+
+        this._duration = duration;
+
+        //Create a HLS object.
+        this.hls = new Hls({
+            debug: debug,
+            startPosition: sourceOffset,
+            maxBufferLength: maxBufferLength,
+            maxMaxBufferLength: duration,
+            backBufferLength: duration
+        });
+
+        this.hls.on(Hls.Events.BUFFER_APPENDED, () => {
+            if (this._element && this._element.buffered.length > 0 && duration !== undefined) {
+                const start = this._element.buffered.start(0);
+                const end = this._element.buffered.end(0);
+                console.debug(`clipId ${this._id}: ${start} - ${end}`);
+
+                if (start <= sourceOffset && end >= sourceOffset + duration) {
+                    this.hls.stopLoad();
+                    this._hlsLoading = false;
+                }
+            }
+        });
+
+        this.hls.on(Hls.Events.BUFFER_EOS, () => {
+            console.debug(`clipId ${this._id}: BUFFER_EOS loading`);
+            if (!this._hlsLoading) {
+                this.hls.startLoad(sourceOffset);
+                this._hlsLoading = true;
+            }
+        });
+
+        this.hls.on(Hls.Events.BUFFER_FLUSHED, () => {
+            if (!this._hlsLoading) {
+                console.debug(`clipId ${this._id}: BUFFER_FLUSHED loading`);
+                this.hls.startLoad(sourceOffset);
+                this._hlsLoading = true;
+            }
+        });
+
+        this.hls.on(Hls.Events.BUFFER_RESET, () => {
+            if (!this._hlsLoading) {
+                console.debug(`clipId ${this._id}: BUFFER_RESET loading`);
+                this.hls.startLoad(sourceOffset);
+                this._hlsLoading = true;
+            }
+        });
+
+        if (debug) {
+            console.debug(
+                `clipId: ${id} startOffset: ${sourceOffset}, duration: ${duration}, maxBufferLength: ${maxBufferLength}`
+            );
+        }
+
+        //Set the source path.
+        this._id = id;
+        this.src = src;
+        this.loaded = false;
+        this._displayName = TYPE;
+        this._elementType = "hls";
+    }
+
+    _load() {
+        if (!this.loaded) {
+            //Create a video element.
+            const video = document.createElement("video");
+            this.hls.attachMedia(video);
+            this.hls.loadSource(this.src);
+            this._element = video;
+
+            this.loaded = true;
+            this._hlsLoading = true;
+            this.hls.startLevel = this.hls.levels.length - 1;
+
+            // add this for debugging video elements
+            let container = document.getElementById("video-debug-container");
```

Comment:
this debug code should be behind the `debug` flag too

Reply:
I'm going to remove this actually and have the app do it's own debugging.